### PR TITLE
Add check of strerror_r return value in CUFile HandleIOError

### PR DESCRIPTION
**Pull request by JanuszL**
Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It adds check of strerror_r return value in CUFile HandleIOError

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     adds a check of strerror_r return value in CUFile HandleIOError
 - Affected modules and functionalities:
     std_cufile.cc
 - Key points relevant for the review:
     error message wording
 - Validation and testing:
     CI test run
 - Documentation (including examples):
     NA

**JIRA TASK**: *[DALI-2174]*

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

File: dali/util/std_cufile.cc [hunk 1]
```diff
@@ -106,7 +106,8 @@ void StdCUFileStream::HandleIOError(int64 ret) const {
   if (ret == -1) {
     std::string errmsg(256, '\0');
     int e = errno;
-    strerror_r(e, &errmsg[0], errmsg.size());
+    auto ret = strerror_r(e, &errmsg[0], errmsg.size());
+    DALI_ENFORCE(ret == 0, make_string("Unknown CUFile error: ", e));
```
Comment (mzient) [inline dali/util/std_cufile.cc:110]: Nitpick: I think that this condition isn't very informative for the reader.
```suggestion
    if (ret != 0) {
      DALI_FAIL(make_string("Unknown CUFile error: ", e));
    }
```

**Review by mzient — APPROVED** [review]

File: dali/util/std_cufile.cc (hunk 1 above)
Comment (JanuszL) [reply dali/util/std_cufile.cc]: Done

**Comment by JanuszL** [comment]
!build

**Review by banasraf — APPROVED** [review]

**Pull request merged**